mllib/src/test/scala/org/apache/spark/mllib/fpm/AssociationRulesSuite.scala

```diff
@@ -84,7 +85,7 @@ class AssociationRulesSuite extends SparkFunSuite with MLlibTestSparkContext {
        [1] 23
      */
     assert(results2.size === 30)
-    assert(results2.count(rule => math.abs(rule.confidence - 1.0D) < 1e-6) == 23)
+    assert(results2.count(rule => rule.confidence ~== 1.0D absTol 1e-6) == 23)
```

Comment:
This seems to fail in Jenkins environment. Did it pass in your local environment, @eugen-prokhorenko ?
```
sbt.ForkMain$ForkError: org.scalatest.exceptions.TestFailedException: Expected 0.6 and 1.0 to be within 1.0E-6 using absolute tolerance.
	at org.apache.spark.mllib.util.TestingUtils$DoubleWithAlmostEquals.$tilde$eq$eq(TestingUtils.scala:80)
	at org.apache.spark.mllib.fpm.AssociationRulesSuite.$anonfun$new$4(AssociationRulesSuite.scala:88)
	at org.apache.spark.mllib.fpm.AssociationRulesSuite.$anonfun$new$4$adapted(AssociationRulesSuite.scala:88)
```

Please make this PR pass the Jenkins.

Reply:
I don't know how I missed it. For usage in place of a predicate there is a less strict version of the operator (`~=` returns `false` without throwing the exception). The patch is underway.

Reply:
fixed

Reply:
Thank you for the swift update!